app/components/Header/index.js

```diff
@@ -115,7 +117,7 @@ class Header extends Component<Props, State> {
   };
 
   render() {
-    const { loggedIn, currentUser } = this.props;
+    const { loggedIn, currentUser, loadingFrontpage } = this.props;
```

Comment:
Since this is a general component, it should be agnostic to the frontpage. So you should call this prop something more generic, f.ex. `loading`.